### PR TITLE
Fix width being ignored when has a value of 0

### PR DESCRIPTION
**Pull request by vjeux**
https://github.com/facebook/css-layout/commit/8f6a96adbcb2b0389f948b8113d8b7cb8991bb2d added a test in isDimDefined that checks if `value > 0.0`, but unfortunately, it did not faithfully port the JavaScript version which is `value >= 0.0`: https://github.com/facebook/css-layout/blob/master/src/Layout.js#L306. Sadly, no test covered this so it went unnoticed.

### CODE REVIEW

File: src/Layout.c [hunk 1]
```diff
@@ -440,7 +440,7 @@ static float getDimWithMargin(css_node_t *node, css_flex_direction_t axis) {
 
 static bool isDimDefined(css_node_t *node, css_flex_direction_t axis) {
   float value = node->style.dimensions[dim[axis]];
-  return !isUndefined(value) && value > 0.0;
+  return !isUndefined(value) && value >= 0.0;
```
Comment (vjeux) [inline src/Layout.c]: This was changed

File: src/__tests__/Layout-test.js [hunk 2]
```diff
@@ -2298,6 +2298,17 @@ describe('Layout', function() {
     );
   });
 
+  it('should layout node with a 0 width', function() {
```
Comment (vjeux) [inline src/__tests__/Layout-test.js]: This was changed

File: src/JavaTranspiler.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ function __transpileToJavaCommon(code) {
     .replace(/getJustifyContent\((.+?)\)/g, '$1.style.justifyContent')
     .replace(/getAlignContent\((.+?)\)/g, '$1.style.alignContent')
     .replace(/isPosDefined\((.+?),\s*(.+?)\)/g, '!isUndefined\($1.style.position[$2]\)')
-    .replace(/isDimDefined\((.+?),\s*(.+?)\)/g, '\(!isUndefined\($1.style.dimensions[dim[$2]]\) && $1.style.dimensions[dim[$2]] > 0.0\)')
+    .replace(/isDimDefined\((.+?),\s*(.+?)\)/g, '\(!isUndefined\($1.style.dimensions[dim[$2]]\) && $1.style.dimensions[dim[$2]] >= 0.0\)')
```
Comment (vjeux) [inline src/JavaTranspiler.js]: This was changed

File: src/CSharpTranspiler.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ function __transpileToCSharpCommon(code) {
     .replace(/getJustifyContent\((.+?)\)/g, '$1.style.justifyContent')
     .replace(/getAlignContent\((.+?)\)/g, '$1.style.alignContent')
     .replace(/isPosDefined\((.+?),\s*(.+?)\)/g, '!isUndefined\($1.style.position[$2]\)')
-    .replace(/isDimDefined\((.+?),\s*(.+?)\)/g, '\(!isUndefined\($1.style.dimensions[dim[$2]]\) && $1.style.dimensions[dim[$2]] > 0.0\)')
+    .replace(/isDimDefined\((.+?),\s*(.+?)\)/g, '\(!isUndefined\($1.style.dimensions[dim[$2]]\) && $1.style.dimensions[dim[$2]] >= 0.0\)')
```
Comment (vjeux) [inline src/CSharpTranspiler.js]: This was changed

**Comment by vjeux** [comment]
@facebook-github-bot import

**Comment by lucasr** [comment]
Looks good to me.

**Comment by vjeux** [comment]
@facebook-github-bot shipit

**Pull request merged**